clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -349,7 +351,16 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
 
         if (numHeaders == 0)
             return new DefaultRecord(sizeInBytes, attributes, offset, timestamp, sequence, key, value, Record.EMPTY_HEADERS);
+        try {
```

Comment:
Sorry, my comment could have been clearer. I was saying that if the exception message was correct, the try block should be narrower. If the exception can be thrown from other places, it's OK for the try block to cover the whole method if we: (1) Update the message, (2) Include the original exception as the cause so that we can figure out where it was thrown from.